app/graphql/mutations/test_result/delete.rb

```diff
@@ -14,8 +14,10 @@ class Delete < BaseMutation
 
       def resolve(args = {})
         profile = scoped_profiles.find(args[:profile_id])
-        test_results = scoped_test_results(args).destroy_all
-        profile.destroy! if profile.external
+        policy_profiles = profile.policy_object&.profiles || [profile]
+        test_results = scoped_test_results(profile_id: policy_profiles)
```

Comment:
This creates a behavior split with profile removal. Profile removal deletes all policy profiles *only* for initial policy profile (`external: false`), whilst this removes policy's test results if *any* of the policy profile id is passed.

I'd suggest to have the same behavior as on policy profile removal: removing all policy test results only when initial profile id is passed.

Reply:
Thanks for catching this! I agree it should behave consistently with profile deletion. Updated and added a test for this scenario.